game-app/game-core/src/main/java/games/strategy/engine/framework/startup/mc/ServerModel.java

```diff
@@ -313,36 +315,58 @@ public synchronized void setLocalPlayerType(final String player, final PlayerTyp
     localPlayerTypes.put(player, type);
   }
 
+  /**
+   * Persists the players mappings to re-use upon a game data change. Used to persist the previous
+   * game's player setting for the game restart if a connection is lost.
+   */
+  public void persistPlayersToNodesMapping() {
+    this.playersToNodesMappingPersisted = true;
+  }
+
+  private void updatePlayersOnGameDataChanged(final GameData data) {
```

Comment:
Nit: private methods should be defined after they are used rather than before their usage: https://github.com/triplea-game/triplea/blob/master/docs/development/reference/code-conventions/java-code-conventions.md#depth-first-method-ordering

To follow depth-first ordering, this method should be moved to be lower in the class file after it's first usage (but not necessarily to the bottom of the class). FWIW: This convention is to help build a consistent expectation of where code is located and to help aid readers to read the class file in one single top-to-bottom scan.

Reply:
Thanks, done.